### PR TITLE
Add carto-cloud-native typings

### PR DESCRIPTION
**Pull request by AdriSolid**
Add new carto-cloud-native typings

### CODE REVIEW

File: packages/react-api/src/types.d.ts [hunk 1]
```diff
@@ -1,25 +1,43 @@
...
+import { MAP_TYPES, API_VERSIONS } from '@deck.gl/carto';
 import { FeatureCollection } from 'geojson';
 
+declare enum DEFAULT_MAPS_URLS {
+  V1 = 'https://{user}.carto.com/api/v1/map',
+  V2 = 'https://maps-api-v2.{region}.carto.com/user/{user}',
+  V3 = 'https://maps-{tenant}'
+}
+
 export type Credentials = {
```
Comment (alasarr) [inline packages/react-api/src/types.d.ts:11]: Let's do something like this, to separate credentials for v1 and v2.
```javascript
interface CredentialsCarto {
  apiVersion: API_VERSIONS.V1 | API_VERSIONS.V2,
  username: string,
  apiKey: string,
  region?: string,
  mapsUrl?: string,
  serverUrlTemplate?: string
}

interface CredentialsCartoCloudNative {
  apiVersion: API_VERSIONS.V3,
  apiBaseUrl: string,
  accessToken?: string,
}

export type Credentials = CredentialsCarto | CredentialsCartoCloudNative;
```

**Review by alasarr — CHANGES_REQUESTED** [review]
Let's make clearer the difference between carto vs cartoCloudNative credentials.

I've also found we have a type like this:
```
export enum SourceTypes {
  SQL = 'sql',
  BIGQUERY = 'bigquery'
}
```
And it's used in some tests. It should use the [MAP_TYPES defined in @carto/deck.gl](https://github.com/CartoDB/carto-deck.gl/blob/carto-cloud-native/modules/carto/src/api/maps-api-common.js#L4)

**Comment by alasarr** [comment]
References to SourceTypes.SQL should be removed

**Review by VictorVelarde — APPROVED** [review]
👍🏻

**Pull request merged**